datalad/support/globbedpaths.py

```diff
@@ -106,18 +109,23 @@ def normalize_hit(h):
                     for sub_pattern in self._get_sub_patterns(pattern):
                         sub_hits = glob.glob(sub_pattern)
                         if sub_hits:
-                            expanded.extend(
-                                sorted(map(normalize_hit, sub_hits)))
+                            partial_hits[pattern] = sorted(
+                                map(normalize_hit, sub_hits))
                             break
-                    # ... but we still want to retain the original pattern
-                    # because we don't know for sure at this point, and it
-                    # won't bother the "install, reglob" routine.
-                    expanded.extend([pattern])
-        return expanded
+                    else:
```

Comment:
Might be me, but I'm confused by the indentation of that `else`. Where is the `if` it belongs to?

Reply:
> Where is the `if` it belongs to?

It belongs to the `for`.  I can switch this to using a sentinel value to register the partial match, if you prefer.

Reply:
Oof. Didn't know that syntax. WTF? :-D

I was legitimately confused since only the inner `if` seemed to make sense, but then this shouldn't work. So - execute always after loop? But how does that work out?
Anyways - now I know and with that it's clear. Thx for enlightening me!